### PR TITLE
[ADDED] Listing KeyValue store and KeyValue store names

### LINKED ISSUE #1075: KV API: list buckets and bucket names
Add `KeyValueStores()` and `KeyValueStoreNames()` for listing KV buckets (similarly to #1008)

### PR DESCRIPTION
**Pull request by piotrpio**
Resolves #1075

### CODE REVIEW

File: kv.go [hunk 1]
```diff
@@ -980,3 +965,73 @@ func (kv *kvs) Status() (KeyValueStatus, error) {
 
 	return &KeyValueBucketStatus{nfo: nfo, bucket: kv.name}, nil
 }
+
+// KeyValueStoreNames is used to retrieve a list of key value store names
+func (js *js) KeyValueStoreNames() <-chan string {
+	ch := make(chan string)
+	ctx, cancel := context.WithTimeout(context.Background(), defaultRequestWait)
```
Comment (wallyqs) [inline kv.go:972]: maybe derive this timeout from the ones that are part of the js context? 
https://github.com/nats-io/nats.go/blob/715a5917c806d2858f6fc80663fbf0910ad58ee3/js.go#L234-L238

Comment (piotrpio) [reply kv.go]: Looking at this, it seems I can drop creating the context altogether, as `l.Next()` has its own context (either from `js.opts.ctx` or from `js.opts.wait`). I changed it, take a look if it looks all right.

File: kv.go [hunk 2]
```diff
@@ -980,3 +965,55 @@ func (kv *kvs) Status() (KeyValueStatus, error) {
 
 	return &KeyValueBucketStatus{nfo: nfo, bucket: kv.name}, nil
 }
+
+// KeyValueStoreNames is used to retrieve a list of key value store names
+func (js *js) KeyValueStoreNames() <-chan string {
+	ch := make(chan string)
+	l := &streamLister{js: js}
+	l.js.opts.streamListSubject = fmt.Sprintf(kvSubjectsTmpl, "*")
```
Comment (wallyqs) [inline kv.go:973]: Do you know if this subject will work with the JS Prefix? Or whether it needs something like `js.apiSubj()` to make it so?

Comment (piotrpio) [reply kv.go:973]: This is the filter subject for `STREAM.LIST` api and points to `$KV.%s.>`. AFAIK, `CreateKeyValue()` creates a stream with interest in this subject (`%s` is the name), so there is no need for `$JS` prefix. Or do I misunderstand it?

Comment (wallyqs) [reply kv.go:973]: ahh right, ok then there is no need for the prefix there.

**Review by wallyqs — APPROVED** [review]
LGTM

**Pull request merged**